mapper/container_ports.go

```diff
@@ -20,26 +21,65 @@ func (n ContainerPorts) FromInternal(data map[string]interface{}) {
 	field.FromInternal(data)
 
 	containers := convert.ToInterfaceSlice(data["containers"])
-	ports := convert.ToInterfaceSlice(data["ports"])
+	annotationPorts := convert.ToInterfaceSlice(data["ports"])
+	annotationsPortsMap := map[string]map[string]interface{}{}
 
-	for i := 0; i < len(ports) && i < len(containers); i++ {
+	// process fields defined via annotations
+	for i := 0; i < len(annotationPorts) && i < len(containers); i++ {
 		container := convert.ToMapInterface(containers[i])
 		if container != nil {
-			portsSlice := convert.ToInterfaceSlice(ports[i])
-			var containerPorts []interface{}
+			portsSlice := convert.ToInterfaceSlice(annotationPorts[i])
+			portMap := map[string]interface{}{}
 			for _, port := range portsSlice {
 				asMap, err := convert.EncodeToMap(port)
 				if err != nil {
 					logrus.Warnf("Failed to convert container port to map %v", err)
 					continue
 				}
 				asMap["type"] = "/v3/project/schemas/containerPort"
-				containerPorts = append(containerPorts, asMap)
+				name, _ := values.GetValue(asMap, "name")
```

Comment:
Will there always be a "name" for the port spec?

Reply:
No, it can be empty for the case when it's just one port. For the case where there is more than one port, the name will always be required.